blockchain/chain/src/store/chain_store.rs

```diff
@@ -876,7 +889,9 @@ pub mod headchange_json {
         fn from(wrapper: HeadChange) -> Self {
             match wrapper {
                 HeadChange::Current(tipset) => HeadChangeJson::Current(TipsetJson(tipset)),
-                HeadChange::Apply(tipset) => HeadChangeJson::Apply(TipsetJson(tipset)),
+                HeadChange::Apply(tipset) => {
+                    HeadChangeJson::Apply(TipsetJson(tipset.last().clone()))
```

Comment:
Should this be `.last()` or `.first()`?

Reply:
Good question.

Reply:
Yes, thanks it's `.first()`. Will fix that.